polaris-service/src/main/java/org/apache/polaris/service/exception/IcebergExceptionMapper.java

```diff
@@ -100,4 +103,27 @@ public Response toResponse(RuntimeException runtimeException) {
     LOGGER.debug("Mapped exception to errorResp: {}", errorResp);
     return errorResp;
   }
+
+  public static boolean isAccessDenied(String errorMsg) {
+    // Corresponding error messages for storage providers Aws/Azure/Gcp
+    // We may want to consider a change to Iceberg Core to wrap cloud provider IO exceptions to
+    // Iceberg ForbiddenException
+    boolean isAccessDenied =
+        errorMsg != null
+            && (errorMsg.toLowerCase(Locale.ENGLISH).contains("access denied")
+                || errorMsg
+                    .toLowerCase(Locale.ENGLISH)
+                    .contains("this request is not authorized to perform this operation")
+                || errorMsg.toLowerCase(Locale.ENGLISH).contains("forbidden"));
```

Comment:
It might be good to explicitly label these strings (e.g. `AWS_ACCESS_DENIED_HINT = "access denied"`) to make future maintenance easier